transformer_engine/pytorch/utils.py

```diff
@@ -169,4 +169,5 @@ def safely_set_viewless_tensor_data(
 
 def cast_if_needed(tensor: torch.Tensor, dtype: torch.dtype) -> torch.Tensor:
     """Cast tensor to dtype"""
-    return tensor if tensor is None or tensor.dtype == dtype else tensor.to(dtype)
+    with torch.enable_grad():
+        return tensor if tensor is None or tensor.dtype == dtype else tensor.to(dtype)
```

Comment:
Hmm, shouldn't we check whether the original tensor required grad before doing this?

Reply:
The behavior is the same. If it didn't require grad then the context manager is a no-op.